### DISCUSSION TITLE
Gif gets corrupted while scaling

### DISCUSSION THREAD

**Discussion by atk91** [ImageMagick/ImageMagick#7471] [Help] [answered]
### ImageMagick version

7.1.1-34

### Operating system, version and so on

macOS

### Description

When I try to resize this gif
![src](https://github.com/ImageMagick/ImageMagick/assets/3193829/e685f23d-8c7b-4550-9701-7dfa685df082)

like so:

```
magick src.gif -scale 400x200 res.gif
```

The `res.gif` looks corrupted
![res](https://github.com/ImageMagick/ImageMagick/assets/3193829/1b6b5602-61de-463a-9e1c-41bba765bdd9)

Why this is happening and how do I fix this?

**UPD:**
Hmm, it seems that it helps to `magick src.gif -coalesce tmp.gif` and then `magick tmp.gif -scale 400x200 res.gif`. Is it intended?

**Comment by snibgo** [comment, marked as answer]
Yes, that is expected.

In your input gif, frames after the first are trimmed to only the pixels that have changed. So the frames are different sizes.
```
magick bird.gif info:

bird.gif[0] GIF 1200x676 1200x676+0+0 8-bit sRGB 256c 0.000u 0:00.000
bird.gif[1] GIF 509x329 1200x676+336+262 8-bit sRGB 256c 0.002u 0:00.001
bird.gif[2] GIF 509x417 1200x676+336+174 8-bit sRGB 256c 0.003u 0:00.002
bird.gif[3] GIF 562x590 1200x676+306+86 8-bit sRGB 256c 129537B 0.004u 0:00.003
```
`-layers coalesce` expands each frame to the full size:
```
magick bird.gif -layers coalesce info:
bird.gif[0] GIF 1200x676 1200x676+0+0 8-bit sRGB 0.000u 0:00.000
bird.gif[1] GIF 1200x676 1200x676+0+0 8-bit sRGB 0.001u 0:00.001
bird.gif[2] GIF 1200x676 1200x676+0+0 8-bit sRGB 0.002u 0:00.002
bird.gif[3] GIF 1200x676 1200x676+0+0 8-bit sRGB 0.003u 0:00.002
```
So, first coalesce, then do the scale. Then, if you want, optimize the frames to again trim them to only the changed pixels.
```
magick bird.gif -layers coalesce -scale 400x200 -layers optimize out.gif
```

![out](https://github.com/ImageMagick/ImageMagick/assets/33812368/6ce275b4-1b79-4d53-b429-05b8ed305de0)
